### DISCUSSION TITLE
frigate.process acceleration?

### DISCUSSION THREAD

**Discussion by grantwest** [blakeblackshear/frigate#2186] [Ask A Question]
It looks like the frigate.process for each camera are the highest CPU load for me. Even higher than ffmpeg and I'm not running any hardware acceleration. frigate.process uses about 0.9 cores per camera where ffmpeg uses about 0.7 without acceleration. What is frigate.process doing? And is there any way to accelerate it?

Right now I'm running on a Skylake Core X Series (no igpu)  and the GPU I do have in the machine is ancient and just used to display the terminal. Right now I have plenty of CPU for the couple of cameras I have, but I am planning on getting a coral and a GPU to get all my CPU back. If I can accelerate the frigate.process then I'd like to make sure I get the right hardware to make that happen.

**Comment by blakeblackshear** [comment]
That process is doing the motion detection, prep for object detection, and tracking. It's CPU usage should go up and down alongside the detection_fps. Watching the debug stream with motion boxes and regions should give you an idea of how much work that process is doing.
